packages/components/select/__tests__/select.test.tsx

```diff
@@ -1179,6 +1179,181 @@ describe("validation", () => {
     user = userEvent.setup();
   });
 
+  describe("validationBehavior=native", () => {
+    it("supports isRequired", async () => {
+      const {getByTestId} = render(
+        <Form data-testid="form">
+          <Select
+            isRequired
+            data-testid="select"
+            label="Test"
+            name="select"
+            validationBehavior="native"
+          >
+            <SelectItem key="one">One</SelectItem>
+            <SelectItem key="two">Two</SelectItem>
+            <SelectItem key="three">Three</SelectItem>
+          </Select>
+        </Form>,
+      );
+
+      const select = getByTestId("select");
+      const input = document.querySelector<HTMLSelectElement>("[name=select]");
+
+      expect(input).toHaveAttribute("required");
+      expect(input?.validity.valid).toBe(false);
+      expect(select).not.toHaveAttribute("aria-describedby");
+
+      act(() => {
+        (getByTestId("form") as HTMLFormElement).checkValidity();
+      });
+
+      expect(input?.validity.valid).toBe(false);
+      expect(select).toHaveAttribute("aria-describedby");
+      expect(document.getElementById(select.getAttribute("aria-describedby")!)).toHaveTextContent(
+        "Constraints not satisfied",
+      );
+
+      await user.click(select);
+      await user.click(document.querySelectorAll("[role='option']")[0]);
+      expect(input?.validity.valid).toBe(true);
+      expect(select).not.toHaveAttribute("aria-describedby");
+    });
+
+    it("supports validate function", async () => {
+      const {getByTestId} = render(
+        <form>
+          <Select
+            data-testid="select"
+            defaultSelectedKeys={["two"]}
+            label="Test"
+            name="select"
+            validate={(value) => (value.includes("two") ? "Invalid value" : null)}
+            validationBehavior="native"
+          >
+            <SelectItem key="one">One</SelectItem>
+            <SelectItem key="two">Two</SelectItem>
+            <SelectItem key="three">Three</SelectItem>
+          </Select>
+          <button data-testid="submit" type="submit">
+            Submit
+          </button>
+        </form>,
+      );
+
+      const select = getByTestId("select");
+      const input = document.querySelector<HTMLSelectElement>("[name=select]");
+
+      expect(input?.validity.valid).toBe(false);
+      expect(select).not.toHaveAttribute("aria-describedby");
+
+      await user.click(getByTestId("submit"));
+      expect(select).toHaveAttribute("aria-describedby");
```

Comment:
Should check for `aria-invalid` on input when a change is expected since input is the actual `<select>` element and select is just the trigger button